File: examples/simple/Python/reader.py
#!/usr/bin/env python3

import adios2
import time
import socket
import sys

class mpi:
    UseComm = True

if mpi.UseComm:
    from mpi4py import MPI


if __name__ == "__main__":

    if mpi.UseComm:
        comm = MPI.COMM_WORLD
        adios = adios2.ADIOS(comm)
    else:
        adios = adios2.ADIOS()
        comm = None

    #@effis-init comm=comm

    #@effis-begin "Jabberwocky"->"Jabberwocky"
    reader_io = ad.DeclareIO("Jabberwocky")

    if mpi.UseComm:
        reader = reader_io.Open("Jabberwocky.bp", adios2.Mode.Read, comm)
    else:
        reader = reader_io.Open("Jabberwocky.bp", adios2.Mode.Read)

    while True:

        status = reader.BeginStep(kittie.Kittie.ReadStepMode, 10.0)

        if status == adios2.StepStatus.NotReady:
            continue
        elif status != adios2.StepStatus.OK:
            break

        print("Found:", reader.CurrentStep()); sys.stdout.flush()
        reader.EndStep()

    reader.Close()
    #@effis-end

    #@effis-finalize

File: OldFrontend/examples/simple/Cpp/reader.cpp
#include "adios2.h"
#include <mpi.h>
#include <thread>
#include <chrono>

int main(int argc, char **argv)
{
	MPI_Comm comm;
	MPI_Init(&argc, &argv);
	MPI_Comm_dup(MPI_COMM_WORLD, &comm);

    //@effis-init comm=comm
	adios2::ADIOS adios(comm);

    //@effis-begin "Jabberwocky"->"Jabberwocky"
	adios2::IO reader_io = adios.DeclareIO("Jabberwocky");
	adios2::Engine reader = reader_io.Open("Jabberwocky.bp", adios2::Mode::Read, comm);

    while (true)
	{
		adios2::StepStatus status = reader.BeginStep(adios2::StepMode::Read, 1.0);

        if (status == adios2::StepStatus::NotReady)
		{
            continue;
		}
		else if (status != adios2::StepStatus::OK)
		{
            break;
		}
		std::cout << "Read step: " << reader.CurrentStep() << std::endl;
        reader.EndStep();
	}

    reader.Close();
    //@effis-end

   //@effis-finalize
}
